matlab/MeCorr/MATLAB_CODE_me.m

```diff
@@ -0,0 +1,318 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+hmei_list = {'BRD-A02303741';'BIX-01294';'methylstat';'QW-BI-011';...
+    'UNC0321';'CBB-1007';'UNC0638';'GSK-J4'};
+hmei_list= cell2table(hmei_list);
+hmei_list.Properties.VariableNames{'hmei_list'}='cpd_name';
+clearvars r_basal
+r_basal(height(hmei_list))= struct('rho',0, 'p',0, 'rlowerbound',0, 'rupperbound',0);
+pp_basal= zeros(height(hmei_list), 3);
+for j = 1:height(hmei_list)
+    fx = find(ismember(ctd2compoundidname_name, hmei_list(j,1)));
+    ix = ismember(drug_auc_expt(:,3), ctd2compoundidname_id(fx,1));
+    sum(ix) % 847. ix is a 1D array. Each drug appears multiple times (appears for each medium in drug_auc_expt)
+    hme_auc_dat = drug_auc_expt(ix,:);
+    
+    [ix, pos] = ismember(hme_auc_dat(:,1), exptidcelllinemediamatch(:,1)); sum(ix); %
+    v1 = hme_auc_dat(:,2); v1= table2array(v1);
+    v2 = grate_ccle_exp_soft(pos,:); % growth rate
+    v3 = fluxstate_gurobi(pos,:);
+    
+    % Scatter plots of methylation (metabolic) flux & calculate correlation
+    figure;
+    scatter(v1, v3)
+    ylabel({'Methylation Flux'},'fontname','helvetica'); %'fontweight','bold')
+    xlabel({'Sensitivity (AUC)'; hmei_list.cpd_name{j}},'fontname','helvetica');
+    ylim([0, 0.06])
+    [R, P, RL, RU]= corrcoef(v1, v3);
+    r_basal(j).rho=R; r_basal(j).p=P; r_basal(j).rlowerbound=RL; r_basal(j).rupperbound=RU;
+    str=sprintf('r= %1.3f',r_basal(j).rho(1,2));
+    T = text(min(get(gca, 'xlim')), max(get(gca, 'ylim')), str);
+    set(T, 'fontsize', 14, 'verticalAlignment', 'top', 'horizontalAlignment', 'left');
+    
+    % Boxplot pairs of methylation flux (Actually grapphing metabolic flux, 
+    ...but we hypothesize met flux approximates methyl flux)
+    ix0 = (v3 < 0.05);sum(ix0); % 0.05 is  the default threshold value
+    ix01 = (v3 > 0.05);sum(ix01);
+    [hh, pp_basal(j,3)] = ttest2(v1(ix0), v1(ix01)); %
+    
+    groups = NaN(size(ix0));
+    groups(ix0) = 1; groups(ix01) = 2;
+    vv = NaN(length(v1), 2);
+    vv(1:sum(groups == 1),1) = v1(groups == 1);
+    vv(1:sum(groups == 2),2) = v1(groups == 2);
+    
+    figure;
+    %clf; UnivarScatter(vv,'Width', 0.3, 'PointSize', 11,'MarkerEdgeColor','w','LineWidth',0.1);%, 'markerfacealpha',0.5);
+    %hold on;
+    bh = boxplot(v1, groups,'symbol','')
+    set(gca,'xticklabel',{'Low growth','High growth'})
+    ylabel({'Sensitivity (AUC)'},'fontname','helvetica');%,'fontweight','bold');
+    xlabel(hmei_list{j,1});
+    ylim([0 20])
+    str=sprintf('p= %1.3f',pp_basal(j,3));
+    T = text(min(get(gca, 'xlim')), max(get(gca, 'ylim')), str);
+    set(T, 'fontsize', 14, 'verticalAlignment', 'top', 'horizontalAlignment', 'left');
+    
+    ix0 = (v3 <= prctile(v3, 25)); sum(ix0);
+    ix01 = (v3 > prctile(v3, 25)); sum(ix01);
+    [hh, pp_basal(j,1)] = ttest2(v1(ix0), v1(ix01));
+    
+    ix0 = (v3 <= prctile(v3, 50)); sum(ix0);

```

Comment:
What does the 50th percentile of v3 mean? That's approximately the median value of the dataset, and you may want to be more stringent, like using 75.